compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirNotImplementedOverrideChecker.kt

```diff
@@ -0,0 +1,278 @@
+                    reporter.reportOn(source, FirErrors.MANY_INTERFACES_MEMBER_NOT_IMPLEMENTED, declaration, representative, context)
+                } else {
+                    reporter.reportOn(source, FirErrors.MANY_IMPL_MEMBER_NOT_IMPLEMENTED, declaration, representative, context)
+                }
+                alreadyReportedManyNotImplemented = true
+*/
+            } else {
+                assert(fakeOverrides.size == 1)
+                val notImplemented = fakeOverrides.single()
+                if ((notImplemented as? FirMemberDeclaration)?.isAbstract != true) continue
+                if (canHaveAbstractMembers || alreadyReportedAbstractNotImplemented) continue
+
+                val notImplementedFunction = notImplemented as? FirSimpleFunction
+                // TODO: suspend function overridden by a Java class in the middle is not properly regarded as an override.
+                if (notImplementedFunction?.isSuspend == true) continue
+                // TODO: Collection extensions are not properly regarded as an override.
+                val originalPackage =
+                    notImplementedFunction?.originalIfFakeOverride()?.dispatchReceiverType?.classId?.packageFqName?.asString()
+                if (originalPackage in COLLECTIONS) continue
+
+                if (notImplemented.isFromInterface(context)) {
+                    reporter.reportOn(source, FirErrors.ABSTRACT_MEMBER_NOT_IMPLEMENTED, declaration, notImplemented, context)
+                } else {
+                    reporter.reportOn(source, FirErrors.ABSTRACT_CLASS_MEMBER_NOT_IMPLEMENTED, declaration, notImplemented, context)
+                }
+                alreadyReportedAbstractNotImplemented = true
+            }
+        }
+    }
+
+    private fun FirCallableDeclaration<*>.isFromInterface(context: CheckerContext): Boolean =
+        (getContainingClass(context) as? FirRegularClass)?.isInterface == true
+
+    private fun collectCallableMembers(firClass: FirClass<*>, context: CheckerContext): Collection<FirCallableDeclaration<*>> =
+        firClass.declarations.filterIsInstance<FirCallableDeclaration<*>>() +
+                dataClassMembers(firClass, context) +
+                delegatedMembers(firClass, context)
+
+    private val FirSimpleFunction.matchesEqualsSignature: Boolean
+        get() = valueParameters.size == 1 && valueParameters[0].returnTypeRef.coneType.isNullableAny &&
+                returnTypeRef.coneType.isBoolean
+
+    private val FirSimpleFunction.matchesHashCodeSignature: Boolean
+        get() = valueParameters.isEmpty() &&
+                returnTypeRef.coneType.isInt
+
+    private val FirSimpleFunction.matchesToStringSignature: Boolean
+        get() = valueParameters.isEmpty() &&
+                returnTypeRef.coneType.isString
+
+    private val FirSimpleFunction.matchesDataClassSyntheticMemberSignatures: Boolean
+        get() = (this.name == EQUALS_NAME && matchesEqualsSignature) ||
```

Comment:
Please use `EQUALS` and `TO_STRING` from `OperatorNameConventions`.

Reply:
Yes, noticed this: https://github.com/JetBrains/kotlin/commit/b138b166e3d5ffba8205d24f238491d912af4a21 Thanks.